.github/workflows/bin/verify-versions.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env spack-python
+import re
+from typing import Dict, List, Union
+
+import llnl.util.filesystem as fs
+from llnl.util import tty
+
+import spack.paths
+import spack.repo
+import spack.util.git
+from spack.version import GitVersion, StandardVersion
+
+BUILTIN = re.compile(r"var\/spack\/repos\/builtin\/packages\/([^\/]+)\/package\.py")
+
+
+def get_modified_files(from_ref: str = "HEAD~1", to_ref: str = "HEAD") -> List[str]:
+    """Get a list of files modified between `from_ref` and `to_ref`
+
+    Args:
+       from_ref (str): oldest git ref, defaults to `HEAD~1`
+       to_ref (str): newer git ref, defaults to `HEAD`
+
+    Returns:
+      files (list): list of file paths
+    """
+    git = spack.util.git.git(required=True)
+
+    with fs.working_dir(spack.paths.prefix):
+        stdout = git("diff", "--name-only", "-r", from_ref, to_ref, output=str)
+
+    return stdout.split()
+
+
+def get_added_versions(
+    checksums_version_dict: Dict[str, Union[StandardVersion, GitVersion]],
+    path: str,
+    from_ref: str = "HEAD~1",
+    to_ref: str = "HEAD",
+) -> List[Union[StandardVersion, GitVersion]]:
+    """Get a list of the versions added between `from_ref` and `to_ref`.
+
+    Args:
+       checksums_version_dict (Dict): all package versions mapped to known checksums.
+       path (str): path to the package.py
+       from_ref (str): oldest git ref, defaults to `HEAD~1`
+       to_ref (str): newer git ref, defaults to `HEAD`
+
+    Returns:
+       versions_list (List): list of versions added between refs
+    """
+    git = spack.util.git.git(required=True)
+
+    # Gather git diff
+    diff_lines = git("diff", from_ref, to_ref, "--", path, output=str).split("\n")
+
+    # Store added and removed versions
+    added_checksums = set()
+    removed_checksums = set()
+
+    # Scrape diff for modified versions
+    for checksum in checksums_version_dict.keys():
+        for line in diff_lines:
+            if checksum in line:
+                if line.startswith("+"):
+                    added_checksums.add(checksum)
+                if line.startswith("-"):
+                    removed_checksums.add(checksum)
+
+    return [checksums_version_dict[c] for c in added_checksums - removed_checksums]
+
+
+def get_standard_version_checksum(pkg, version: StandardVersion) -> str:
+    """Get a checksum for a Spack StandardVersion.
+
+    Args:
+       pkg (Package): Spack package in for which to get a version checksum
```

Comment:
Nit. The description is a bit awkward:

```suggestion
       pkg (Package): Spack package for which to get a version checksum
```

Reply:
Good catch! Accidentally left `in` within the sentence after changing it.